river/drift/hddm_a.py

```diff
@@ -12,7 +11,7 @@ class HDDM_A(DriftDetector):
     values and returns the estimated status of the stream: STABLE, WARNING or
     DRIFT.
 
-    **Input:** `value` must be a binary signal, where 1 indicates error.
+    **Input:** `x` must be a binary signal, where 1 indicates error.
```

Comment:
Same here. But now that I read this again, I don't think I understand the explanation. As in, what's an error? 😅

Reply:
As far as I understand, these detectors were developed having a classifier in mind.

They always monitor a Bernoulli random variable, where 0 indicates that a classifier has not made a mistake.

The authors of the paper [Combining univariate approaches for ensemble change detection in multivariate data](https://www.sciencedirect.com/science/article/abs/pii/S1566253517301239) place EWMA, EDM, EDDM, HDDM* under the category of "Control Charts". These are the ones that have three states:

- control
- warning
- drift

Some, more recent, drift detectors such as ADWIN and KSWIN can work with continuous data. Page-Hinckley also accepts continuous data. The three of them only have a "normal" and "drift" state.

Ideally, we could differentiate these detectors via different base classes, but I don't know how to keep allowing them to be used in the ensembles and trees, due to the different types of inputs.

The easy solution is to keep things as they are and focus on documenting the update method.

Pinging @MaxHalford and @jacobmontiel!

Reply:
> The easy solution is to keep things as they are and focus on documenting the update method.

Sounds good to me!